src/js/send-web-vitals.js

```diff
@@ -1,7 +1,48 @@
 function sendWebVitals() {
 
-  function sendWebVitalsGAEvents({name, delta, id, attribution, navigationType}) {
+  function getLoafAttribution(attribution) {
+    if (!attribution) {
+      return {};
+    }
+
+    const entry = attribution.eventEntry;
+
+    if (!entry) {
+      return {};
+    }
+
+    const loafAttribution = {
+      total: 0
+    };
+
+    const longAnimationFrames = performance.getEntriesByType('long-animation-frame');
+    longAnimationFrames.filter(loaf => {
+      // LoAFs that intersect with the event.
+      return entry.startTime < (loaf.startTime + loaf.duration) && loaf.startTime < (entry.startTime + entry.duration);
+    }).forEach(loaf => {
+      loaf.scripts.forEach(script => {
```

Comment:
I wonder if we should also capture the overall LoAF timings?: https://gist.github.com/noamr/2d4e2bdd48661fa39e32283efecaf418#file-whynp-js-L10-L17

And should we put the script stuff in it's own `longestScript` object within the `loafAttribution` object to more accurately reflect what it is showing?

It would also be good to generalise this into a `summariseLoAF` function so could also capture all LoAFs separately with a PO, as an alternative approach to compare whether targeting INP LoAF or LoAFs in general is most useful:

```js
  // Monitor LoAFs above a 150ms threshold
  if (PerformanceObserver.supportedEntryTypes.includes('long-animation-frame')) {
    new PerformanceObserver(entries => {
      for (const entry of entries.getEntries()) {
        if (entry.duration > 150) {
          gtag('event', 'LoAF', summariseLoAF(entry));
        }
      }
    }).observe({type: 'long-animation-frame', buffered: true});
  }
```

Reply:
Done